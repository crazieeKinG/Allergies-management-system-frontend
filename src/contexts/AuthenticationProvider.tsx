import React, { createContext, useState } from "react";
import AuthenticationContextInterface, {
    AuthenticationContextDataInterface,
} from "../interfaces/authentication.interfaces";

interface Props {
    children: React.ReactNode;
}

export const AuthenticationContext = createContext<
    AuthenticationContextInterface | undefined
>(undefined);

const AuthenticationProvider = ({ children }: Props) => {
    const initialAuthenticationData: AuthenticationContextDataInterface = {
        username: "Saajan Shrestha",
        accessToken:
            "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOiJiMDVlYWE3YS1iYThiLTQxYzItYmViZC00N2I0ODRmMTUwYmIiLCJpYXQiOjE2NjQ0MjI5Nzh9.QZZc2h3noQYWnvrTRDHjj_fvhKw3Ri-vApTz6E1KIuw",
    };
    const [authentication, setAuthentication] = useState(
        initialAuthenticationData
    );

    return (
        <AuthenticationContext.Provider
            value={{ authentication, setAuthentication }}
        >
            {children}
        </AuthenticationContext.Provider>
    );
};

export default AuthenticationProvider;
