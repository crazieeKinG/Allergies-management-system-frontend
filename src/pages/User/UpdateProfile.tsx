import { Alert, Col, Divider, Skeleton, Space, Typography } from "antd";
import moment from "moment";
import { useEffect, useState } from "react";
import { getUserData } from "../../api/User/user.api";
import SignupForm from "../../components/Signup/SignupForm";
import { DEFAULT_ALERT_VALUE } from "../../constants/alert.constants";
import AlertMessageInterface from "../../interfaces/alert.interfaces";
import UserInterface from "../../interfaces/user.interfaces";

const UpdateProfile = () => {
    const [defaultValue, setDefaultValue] = useState<UserInterface | undefined>(
        undefined
    );

    const [alertMessage, setAlertMessage] =
        useState<AlertMessageInterface>(DEFAULT_ALERT_VALUE);

    useEffect(() => {
        getUserData()
            .then((response) => {
                setDefaultValue({
                    ...response.data,
                    dateOfBirth: moment(response.data.dateOfBirth),
                });
            })
            .catch((error) => {
                if (!error)
                    setAlertMessage({
                        type: "error",
                        message:
                            "Cannot connect to the server. Please try again later.",
                    });
                else
                    setAlertMessage({
                        type: "warning",
                        message: error.message,
                    });
            });
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);

    return (
        <>
            {alertMessage.message && (
                <Alert
                    type={alertMessage.type}
                    message={alertMessage.message}
                />
            )}
            {!defaultValue ? (
                <Skeleton />
            ) : (
                <Space direction="vertical" size={20} className="d-flex">
                    <Col span={24}>
                        <Typography.Title>Update user profile</Typography.Title>
                        <Divider />

                        <Col span={12}>
                            <SignupForm initialValue={defaultValue} />
                        </Col>
                    </Col>
                </Space>
            )}
        </>
    );
};

export default UpdateProfile;
