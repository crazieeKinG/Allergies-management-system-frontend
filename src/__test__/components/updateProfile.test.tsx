import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import AuthenticationProvider from "../../contexts/AuthenticationProvider";
import UpdateProfile from "../../pages/User/UpdateProfile";
import "../mocks/matchMedia";

describe("Update Profile", () => {    
    it("should generate update profile page", () => {
        const view = render(
            <AuthenticationProvider>
                <BrowserRouter>
                    <UpdateProfile />
                </BrowserRouter>
            </AuthenticationProvider>
        );
        expect(view.asFragment()).toMatchSnapshot();
    });
});
