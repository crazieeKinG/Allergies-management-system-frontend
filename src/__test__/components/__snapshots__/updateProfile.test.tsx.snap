// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Update Profile should generate update profile page 1`] = `
<DocumentFragment>
  <div
    class="ant-skeleton"
  >
    <div
      class="ant-skeleton-content"
    >
      <h3
        class="ant-skeleton-title"
        style="width: 38%;"
      />
      <ul
        class="ant-skeleton-paragraph"
      >
        <li />
        <li />
        <li
          style="width: 61%;"
        />
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
